ephios/extra/auth.py

```diff
@@ -44,26 +49,50 @@ def update_user(self, user, claims):
             except ValueError:
                 pass
         user.save()
-        if self.provider.group_claim:
-            groups = set(self.provider.default_groups.all())
-            groups_in_claims = (
-                reduce(
-                    lambda d, key: d.get(key, None) if isinstance(d, dict) else None,
-                    self.provider.group_claim.split("."),
-                    claims,
+        self._update_user_groups(user, claims)
+        self._update_user_qualifications(user, claims)
+        oidc_update_user.send(self, user=user, claims=claims, provider=self.provider)
+        return user
+
+    def _update_user_qualifications(self, user, claims):
+        if not self.provider.qualification_claim:
+            return
+        target_qualification_uuids = []
+        for codename in dotted_get(claims, self.provider.qualification_claim, []):
+            try:
+                target_qualification_uuids.append(
+                    uuid.UUID(
+                        str(self.provider.qualification_codename_to_uuid.get(codename, codename))
+                    )
                 )
-                or []
+            except ValueError:
+                pass
+
+        target_qualifications = Qualification.objects.filter(uuid__in=target_qualification_uuids)
```

Comment:
this makes the behaviour a bit inconsistent between qualifications (uuid-based) and groups (name-based). not sure if thats necessarily bad, just maybe not so intuitive

Reply:
Let's see how it goes. I think I'd prefer some codename for Group names as well, so as the human readable group name can be changed freely and can use emoji etc whereas the claim can use slug-like codenames.